Documentation/Design/Lambdas/Lambda Declare.md

```diff
@@ -0,0 +1,162 @@
+# Lambda Declare
+
+## Overview
+GLS will support declaring lambda functions. In order to declare a function that takes a lambda function as a parameter, the lambda must first be declared.
+
+## Commands
+
+### `lambda declare`
+`lambda declare : lambdaTypeName lambdaFunctionName returnType *`[parameterName parameterType ...]`*
+
+## Usage:
+
+```gls
+lambda declare : over21 check boolean age int
```

Comment:
Can you have something like `overMinimum` (without the numbers in the name)? Having the `21` there feels weird, particularly with casing.